New custom 404 page

dmundra opened the pull request:
@garaynakca here is the draft pull request for the 404 page. The branch is called `new-404-page`.

<!-- readthedocs-preview civicactions-handbook start -->
----
:books: Documentation preview :books:: https://civicactions-handbook--1043.org.readthedocs.build/en/1043/

<!-- readthedocs-preview civicactions-handbook end -->

dmundra committed:
Added 404 page override.

garaynakca committed:
Update 404.html

Added new copy to the 404 page

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dmundra commented on .config/theme_overrides/404.html:
```diff
@@ -0,0 +1,29 @@
+<h3>
+  The page you are looking for has been removed, is temporarily unavailable or
+  does not exist. Maybe try the address again, or head to the homepage.
+</h3>
```
Preview https://civicactions-handbook--1043.org.readthedocs.build/en/1043/not-found

I think having a header for all that text is not good practice for accessibility. Are you looking to make the text larger?

garaynakca replied:
@dmundra Yes.. I was just hoping to bump up the text size.

garaynakca replied:
.... but, it's okay if it has to be the standard body size.

dmundra replied:
You can wrap that in a div and give it a unique class. Then CSS can be added to this file https://github.com/CivicActions/guidebook/blob/new-404-page/assets/css/extra.css to style it. Do you want to try that?

garaynakca replied:
Thanks for CSS guidance!  Will try that!

garaynakca committed:
Update extra.css

garaynakca committed:
Update 404.html

Added .leadtext class to CSS at 1.2rem

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dmundra commented on assets/css/extra.css:
```diff
@@ -23,6 +23,13 @@ font-size of other elements in the theme.
   color: #a9a9a9 !important;
 }
+
+/* adding a larger type option for lead text
```
@garaynakca looks like you missed adding the closing comment `*/` to this line. I think the CSS won't work till that is added.

garaynakca committed:
Update extra.css

Fixed comment with closing */

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dmundra committed:
Merge branch 'master' into new-404-page

garaynakca committed:
Update 404.html

adding new copy

garaynakca committed:
Update 404.html

Fixed tag </h2> 
Added homepage href

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

garaynakca committed:
Update 404.html

Need to add:  <a
      href="{{ config.extra.homepage | d(nav.homepage.url, true) | url }}"
      title="{{ config.site_name | e }}"
      class="md-header__button md-logo"
      aria-label="{{ config.site_name }}"
      data-md-component="logo"
    >

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

404 copy hierarchy

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update extra.css

added A23737 color to lead text

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dmundra commented on .config/theme_overrides/404.html:
```diff
@@ -0,0 +1,30 @@
+{% extends "main.html" %}
+
+<!-- Content -->
+<div class="leadtext">404</div>
```
@garaynakca looks like we lost the block `{% block content %}` to go above this div.

dmundra replied:
I believe that is causing the errors.

garaynakca replied:
Okay... thanks!  Fixing that.

garaynakca committed:
Update 404.html

garaynakca committed:
Update extra.css

.leadtext for 404

garaynakca committed:
Update 404.html

leadtext

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

garaynakca committed:
Update extra.css

changed .leadtext to test

garaynakca committed:
Update 404.html

garaynakca committed:
Update extra.css

font change

grugnog committed:
Merge branch 'master' into new-404-page

garaynakca committed:
Update extra.css

404 text color CA red (not background color)

garaynakca committed:
Update 404.html

adding homepage link (i think)

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update extra.css

garaynakca committed:
Update 404.html

garaynakca committed:
Update 404.html

garaynakca committed:
Update 404.html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update 404.html

Trying to decrease line spacing between 404 and other content

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

garaynakca committed:
Update extra.css

garaynakca committed:
Update 404.html

garaynakca committed:
Update extra.css

font weight for leadtext

garaynakca committed:
Update extra.css

font weight 100 .leadtext

dmundra committed:
Merge branch 'master' into new-404-page

dmundra commented on .config/theme_overrides/404.html:
```diff
@@ -0,0 +1,34 @@
+  <a href="{{ config.extra.homepage | d(nav.homepage.url, true) | url }}">
+    homepage.
+  </a>
+</h2>
```
@garaynakca I think using headers for this one is not best practice. I think '404' should be the h1 and the other two should be divs with styling you would like to see.

garaynakca committed:
Update 404.html (#1087)

* Update 404.html

updated headers

* Update 404.html (#1098)

* Update 404.html

Changed h2 to div text

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dmundra committed:
Merge branch 'master' into new-404-page
